pallets/pallet-bonded-coins/src/curves.rs

```diff
@@ -8,6 +8,26 @@ use substrate_fixed::{
 	types::I9F23,
 };
 
+pub trait ParameterValidation {
+	fn are_parameters_valid(&self) -> bool;
```

Comment:
This somewhat goes against Rust idiomatic guidelines. Ideally you should not have an instance of `&self` if this check would return `false`. You should implement something like `TryFrom` in this case.